### PR TITLE
Hover-effect directive for bootstrap dropdown

### PR DESCRIPTION
**Pull request by tahmeedshah**

### CODE REVIEW

**Review by blaisefonseca — APPROVED** [review]

File: app/components/scbd-angularjs-controls/form-control-directives/hover-effect.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+define(['app','lodash','angular'], function(app,_,angular) {
+    app.directive('hoverEffect',['$timeout', function ($timeout  ) {
+    return {
+        scope: {
+           binding: '=ngModel',
+        },
+        restrict: 'EA',
```
Comment (sbilodeau) [inline app/components/scbd-angularjs-controls/form-control-directives/hover-effect.js:7]: most probably should be only a `A` directive (not an `E`)

File: app/components/scbd-angularjs-controls/form-control-directives/hover-effect.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+define(['app','lodash','angular'], function(app,_,angular) {
+    app.directive('hoverEffect',['$timeout', function ($timeout  ) {
+    return {
+        scope: {
+           binding: '=ngModel',
```
Comment (sbilodeau) [inline app/components/scbd-angularjs-controls/form-control-directives/hover-effect.js:5]: unused

File: app/components/scbd-angularjs-controls/form-control-directives/hover-effect.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        },
+        restrict: 'EA',
+        link: function (scope, element) {
+            element.on( 'mouseenter', function () {
+                if (window.innerWidth > 800) {
+                    element.children("ul").addClass("show");
```
Comment (sbilodeau) [inline app/components/scbd-angularjs-controls/form-control-directives/hover-effect.js:11]: I think you should pass the the query selection into the value of the directive attribute to make it more generic. currently it just works for UL.

eg:
```
<div hover-effect="div.help">
this is my regular content with a <a href="http://...">link to something</a>
<div class="help">
 the content shown only on hover 
</div>

**Review by sbilodeau — CHANGES_REQUESTED** [review]
I think the hover-effect can be easily achieved using CSS classes:

https://jsfiddle.net/tnj8vb7k/9/

```
<div hover-effect>
always shown  content

 <div hover-effect-toggle>
   shown only when hover
 </div>
 
 <div> text after </div>
</div>

<style>

[hover-effect] [hover-effect-toggle]{
    display: none;
}

[hover-effect]:hover  [hover-effect-toggle]{
    display:block;
}

</style>

```
eg: 
https://stackoverflow.com/questions/34459841/show-child-element-on-parent-hover-in-css/34459870

you can also use css transition to make the 500ms delay effect 
https://stackoverflow.com/questions/3331353/transitions-on-the-css-display-property

**Comment by blaisefonseca** [comment]
@tahmeedshah I spoke with @sbilodeau about the css effect, I had read about issues on small screen so we will keep the js part, can you implement his other implementations please?

File: app/components/scbd-angularjs-controls/form-control-directives/hover-effect.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+define(['app','lodash','angular'], function(app,_,angular) {
+    app.directive('hoverEffect',['$timeout', function ($timeout  ) {
+    return {
+        scope: {
+            hoverEffect: '@'
+        },
+        restrict: 'A',
+        link: function (scope, element) {
+            if(scope.hoverEffect==undefined || scope.hoverEffect.length==0) return;
```
Comment (sbilodeau) [inline app/components/scbd-angularjs-controls/form-control-directives/hover-effect.js:9]: `==` with `undefined` is not a good comparaison.  you need `===` to make good comparaison with undefined

Checking if `scope.hoverEffect` is falsy can do the job and easier to read
```suggestion
            if(!scope.hoverEffect) return;
```

**Review by sbilodeau — CHANGES_REQUESTED** [review]
small comment about `==undefined`

**Review by sbilodeau — APPROVED** [review]

**Pull request merged**